ledger/src/blockstore.rs

```diff
@@ -4068,7 +4067,7 @@ pub fn create_new_ledger(
     let blockstore = Blockstore::open_with_options(
         ledger_path,
         BlockstoreOptions {
-            access_type,
+            access_type: AccessType::PrimaryOnly,
```

Comment:
The direction of this PR looks good and makes sense to me, just to double-check whether there's any corner case that I might miss.

I think it shouldn't be the case, but do we have any use case where we just create a new ledger and then close it without writing anything?

Reply:
>but do we have any use case where we just create a new ledger and then close it without writing anything?

Good thing to double check, but no, there are no instances where this would be the case. A little farther down in the function, we actually do some operations that require Primary access:
```
    let blockstore = Blockstore::open_with_options(
        ledger_path,
        BlockstoreOptions {
            access_type: AccessType::PrimaryOnly,
            recovery_mode: None,
            enforce_ulimit_nofile: false,
            shred_storage_type: shred_storage_type.clone(),
        },
    )?;

    ...

    blockstore.insert_shreds(shreds, None, false)?;
    blockstore.set_roots(std::iter::once(&0))?;
    // Explicitly close the blockstore before we create the archived genesis file
    drop(blockstore);
```